ext/imap/php_imap.c

```diff
@@ -790,11 +802,22 @@ PHP_FUNCTION(imap_open)
 	 * the CL_EXPUNGE flag which will expunge when the mailbox is closed (be that manually, or via the
 	 * IMAP\Connection object being destroyed naturally at the end of the PHP script */
 	if (flags && ((flags & ~(OP_READONLY | OP_ANONYMOUS | OP_HALFOPEN | PHP_EXPUNGE | OP_DEBUG | OP_SHORTCACHE
-	 		| OP_SILENT | OP_PROTOTYPE | OP_SECURE)) != 0)) {
+	 		| OP_SILENT | OP_PROTOTYPE | OP_SECURE | PHP_XOAUTH2)) != 0)) {
 		zend_argument_value_error(4, "must be a bitmask of the OP_* constants, and CL_EXPUNGE");
```

Comment:
The error message needs to be adjusted.

Reply:
The constant is called "OP_XAUTH2" when used in PHP so I think the error message is ok, or am I missing the point?

All the other flags here are defined in c-client, so I named the new one PHP_XOAUTH2 internally but kept the OP_ prefix in the interface for consistency. I can see that this is perhaps somewhat confusing when reading the code; maybe we should use another prefix?

Reply:
Ah indeed, it didn't register with me that the userland constant is named OP_XOAUTH2.

Tbh, I'm not sure if using an other prefix makes sense, *maybe* ``IMAP_XOAUTH2``? But then it's the only one following this pattern.